src/plugins/data_source/server/plugin.ts

```diff
@@ -155,7 +165,8 @@ export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourc
     cryptography: CryptographyServiceSetup,
     logger: Logger,
     auditTrailPromise: Promise<AuditorFactory>,
-    authRegistryPromise: Promise<IAuthenticationMethodRegistery>
+    authRegistryPromise: Promise<IAuthenticationMethodRegistery>,
+    customApiSchemaRegistryPromise: Promise<CustomApiSchemaRegistry>
```

Comment:
nit:
Thanks for the change. 

Do we also want to create a ICustomApiSchemaRegistry type similar to [IAuthenticationMethodRegistery](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/55443f7fe3c22afe541c7225f0d25207f1b03a77/src/plugins/data_source/server/auth_registry/authentication_methods_registry.ts#L9) to only expose ```register()``` also make the format consistent if not complex?